packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerHighlight/utils/common/textSpanUtils.ts

```diff
@@ -0,0 +1,58 @@
+import { TextSpan } from '../../types';
+
+export const START = 0;
+export const END = 1;
+
+export function spanGetText<T extends string | null | undefined>(text: T, span: TextSpan) {
+  if (!text) return text;
+  if (spanLen(span) === 0) return '';
+  return text.substring(span[START], span[END]);
+}
+
+export function spanLen(span: TextSpan) {
+  return Math.max(0, span[END] - span[START]);
+}
+
+export function spanIntersects([beginA, endA]: TextSpan, [beginB, endB]: TextSpan): boolean {
```

Comment:
> There already exists a function (with a more correct name) of spansIntersect in documentUtils.ts. Let's only have a single version.

I made the new method because of the difference of the type of arguments. I can refactor the new function to use the existing  `spansIntersect`. However, I just found the existing one is having off-by-one error and not sure the impact of fixing it. So, I tentatively added a TODO comment. Would you have any suggestion?

Reply:
I tried to fix the existing `spansIntersect`, but it resulted in test error. Need to fix it outside of this PR and revisit. I added comment.